### PR TITLE
Jenkins automated tests

### PR DESCRIPTION
**Pull request by Erikvv**
Add a Jenkinsfile to start storj-sim and run static analysis (phpstan) and automated tests.

### CODE REVIEW

**Comment by Erikvv** [comment]
I finally got the PHPUnit tests working in Jekins.

**Comment by Erikvv** [comment]
uplink-c and some other things need a version bump but I don't like big PR's so I'll do it in a new PR.

**Comment by stefanbenten** [comment]
@Erikvv In order to make progress on this, should we merge this in order to get the other PRs tested?

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            agent {
+                docker {
+                    image docker.build("storj-ci", "--pull https://github.com/storj/ci.git").id
+                    args '--user root:root --cap-add SYS_PTRACE -v "/tmp/gomod":/go/pkg/mod '
+                }
+            }
+            steps {
+                script {
+                    // need to keep deleting this directory because of some cleanup issue
+                    sh 'rm -rf tmp-c'
```
Comment (stefanbenten) [inline Jenkinsfile:18]: We should move this into the cleanup step that is always executed.

**Review by stefanbenten — COMMENTED** [review]
We probably should not commit the libuplink.so file into the repo.
I'll check with @kaloyan-raev and @mniewrzal about their preference.

File: Jenkinsfile (hunk 1 above)
Comment (Erikvv) [reply Jenkinsfile:18]: `cleanWs()` already does this.

I am not sure why but there are somtimes permission issues during cleanup.

`java.nio.file.FileSystemException: /var/lib/jenkins/workspace/uplink-php_PR-6/tmp-c/error.go: Operation not permitted`

We can move this line but I fear the issue will return.

**Comment by Erikvv** [comment]
> We probably should not commit the libuplink.so file into the repo.
> I'll check with @kaloyan-raev and @mniewrzal about their preference.

I already responded to this feedback here https://github.com/storj-thirdparty/uplink-php/pull/1#issuecomment-661433232

It is not possible to upload an artifact to Packagist, it needs to be in git.

If you really want this, I can think of a few workarounds:
- Have 1 git repo with the PHP source and 1 with the PHP source + artifact, and publish the latter on packagist.
- Have the user execute a manual build step 
- Publish it on your own composer repository instead of Packagist. For example https://docs.gitlab.com/ee/user/packages/composer_repository/ . The user (developer) needs to add the url to the composer.json file in their project.

**Comment by Erikvv** [comment]
> We probably should not commit the libuplink.so file into the repo.

I presume the same would go for libuplink.h because they are both build artifacts?

**Comment by Erikvv** [comment]
> @Erikvv In order to make progress on this, should we merge this in order to get the other PRs tested?

I guess you mean that the tests added in #2 and #4 are currently not run in Jenkins.

So yes if you want to achieve that then merge this first. I can also rebase them on top of this PR.

**Comment by Erikvv** [comment]
All changes in this PR are moved to #8

**Pull request closed without merging**